system-tests/e2e-testsupport/src/main/java/com/hotels/styx/servers/MockOriginServer.java

```diff
@@ -0,0 +1,218 @@
+ * limitations under the License.
+ */
+package com.hotels.styx.servers;
+
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.client.RequestPatternBuilder;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.client.UrlMatchingStrategy;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockApp;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.global.NotImplementedRequestDelayControl;
+import com.github.tomakehurst.wiremock.http.AdminRequestHandler;
+import com.github.tomakehurst.wiremock.http.BasicResponseRenderer;
+import com.github.tomakehurst.wiremock.http.ProxyResponseRenderer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.RequestHandler;
+import com.github.tomakehurst.wiremock.http.StubRequestHandler;
+import com.github.tomakehurst.wiremock.http.StubResponseRenderer;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ServiceManager;
+import com.hotels.styx.api.HttpHandler2;
+import com.hotels.styx.server.HttpConnectorConfig;
+import com.hotels.styx.server.HttpServer;
+import com.hotels.styx.server.HttpServers;
+import com.hotels.styx.server.HttpsConnectorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.configureFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.google.common.base.Optional.absent;
+import static com.hotels.styx.servers.WiremockResponseConverter.toStyxResponse;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.emptyMap;
+import static rx.Observable.just;
+
+public final class MockOriginServer {
+    private static final String FILES_ROOT = "__files";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MockOriginServer.class);
+    private static final int MAX_CONTENT_LENGTH = 256 * 1024;
+
+    private final String appId;
+    private final String originId;
+    private final int adminPort;
+
+    private int serverPort;
+    private final HttpServer adminServer;
+    private final HttpServer mockServer;
+
+    static {
+        System.setProperty("org.mortbay.log.class", "com.github.tomakehurst.wiremock.jetty.LoggerAdapter");
+    }
+
+    private ServiceManager services;
+
+    private MockOriginServer(String appId, String originId, int adminPort, int serverPort, HttpServer adminServer, HttpServer mockServer) {
+        this.appId = appId;
+        this.originId = originId;
+        this.adminPort = adminPort;
+        this.serverPort = serverPort;
+        this.adminServer = adminServer;
+        this.mockServer = mockServer;
+    }
+
+    private static WireMockApp wireMockApp() {
```

Comment:
Could we move this method further down the class? I think it makes more sense for public methods to be the first thing the viewer sees, as they convey how the class can be used.

Reply:
Ok. I have moved all private methods to the end of the class.